torch/csrc/jit/script/compiler.cpp

```diff
@@ -1499,11 +1553,27 @@ struct to_ir {
 
   std::shared_ptr<SugaredValue> emitApplyExpr(Apply &apply, size_t n_binders) {
     auto sv = emitSugaredExpr(apply.callee(), 1);
-    auto inputs = getNamedValues(apply.inputs(), true);
     auto attributes = fmap(apply.attributes(), [&](const Attribute& attr) {
       return NamedValue(attr.range(), attr.name().name(), emitExpr(attr.value()));
     });
-    return sv->call(apply.callee().range(), method, inputs, attributes, n_binders);
+
+    auto loc = apply.callee().range();
+    if (sv->kind() == "fork") {
+      JIT_ASSERT(attributes.size() == 0);
+      JIT_ASSERT(n_binders == 1);
+      auto& trees = apply.inputs().tree()->trees();
+      if (trees.size() != 2) {
```

Comment:
I think it would be nicer if the syntax of fork were:

``` torch.fork(<required_function_target>, arg1, arg2, name=kwargforfunction)```

That way kwargs would work and it would reduce the typechecking burden here.